### ISSUE TITLE
Envoy 1.12.0 serializes timestamp/duration fields as strings in Zipkin v2 formatted tracing spans

### ISSUE THREAD

**Issue by cetanu** [envoyproxy/envoy#8871]
**tl;dr:** timestamp/duration are emitted as strings for the v2 zipkin format - this thread tries to determine if this is a bug

Firstly, thank you for the work to get the v2 format supported in the first place.

I just tried reading through the source for the zipkin tracer extension and to be honest wasn't able to fully understand how it all works together.

From what I can tell, the timestamp and duration are added to the span as u64 and i64 respectively which are the correct types according to the zipkin v2 protobuf that I found, but somewhere between then and the serialization, they become strings. 

This is the part that I am confused about. I wanted to submit a PR but I couldn't piece it together.

Here's an example of my tracing configuration:

```
tracing:
  http:
    name: envoy.zipkin
    config:
      collector_cluster: tracing-sidecar
      collector_endpoint: /api/v2/spans
      collector_endpoint_version: HTTP_JSON
      shared_span_context: false
```

and this is the resulting span:

```
[
  {
    "traceId": "83713138fa14c726",
    "id": "83713138fa14c726",
    "kind": "SERVER",
    "name": "www.hello",
    "timestamp": "1572849988488551",  <-------------------------------------------
    "duration": "7131",        <-------------------------------------------
    "localEndpoint": {
      "serviceName": "envoy-cluster",
      "ipv4": "172.31.0.5"
    },
    "tags": {
      "response_flags": "-",
      "request_size": "0",
      "response_size": "292",
      "node_id": "envoy",
      "http.status_code": "200",
      "http.method": "GET",
      "upstream_cluster": "testcluster",
      "guid:x-request-id": "3ab0fc56-f92e-9c03-a087-1f23662699cb",
      "zone": "us-east-1",
      "http.protocol": "HTTP/1.1",
      "component": "proxy",
      "http.url": "https://www.hello/headers",
      "downstream_cluster": "-",
      "user_agent": "HTTPie/0.9.2"
    }
  }
]
```

If this is intentional then that's fine, we'll do some work to cast the strings to what we need.

**Comment by dio** [comment]
@cetanu I think this is related to t way we serialize protobuf to json, which relies on MessageToJsonString. https://github.com/protocolbuffers/protobuf/issues/2679

We could use low level lib e.g. rapidjson to serialize it correctly but it is decided that we stick to protobuf library for doing so.

**Comment by danielgblanco** [comment]
I appreciate this was just a question, but is there an issue reported for this or anywhere else to follow up? This issue blocks anyone from using Zipkin in `HTTP_JSON` version.

**Comment by dio** [comment]
> but is there an issue reported for this or anywhere else to follow up?

@danielgblanco seems like there is no reported issue yet. If you have time could you submit an issue regarding this? I'll try to come up with a proposal to fix this.

**Comment by danielgblanco** [comment]
Brilliant, I'll do that ASAP.

**Comment by danielgblanco** [comment]
Sorry for the delay, I've created https://github.com/envoyproxy/envoy/issues/9341
